diff --git a/msvc/include/sys/time.h b/msvc/include/sys/time.h
index fe7fd84..08c6820 100644
--- a/msvc/include/sys/time.h
+++ b/msvc/include/sys/time.h
@@ -26,3 +26,7 @@
  */
 
 #include <time.h>
+
+#if defined(_MSC_VER) || defined(__clang__)
+#include <winsock.h>
+#endif
diff --git a/src/dvdnav_internal.h b/src/dvdnav_internal.h
index 8bc5706..9f93488 100644
--- a/src/dvdnav_internal.h
+++ b/src/dvdnav_internal.h
@@ -233,8 +233,8 @@ DVDNAV_API int64_t dvdnav_convert_time(const dvd_time_t *time);
     do { if (this) snprintf(this->err_str, MAX_ERR_LEN, format, ## args); } while (0)
 #else
 #ifdef _MSC_VER
-#define printerrf(str) \
-    do { if (this) snprintf(this->err_str, MAX_ERR_LEN, str); } while (0)
+#define printerrf(...) \
+    do { if (this) snprintf(this->err_str, MAX_ERR_LEN, __VA_ARGS__); } while (0)
 #else
 #define printerrf(...) \
     do { if (this) snprintf(this->err_str, MAX_ERR_LEN, __VA_ARGS__); } while (0)
diff --git a/src/read_cache.c b/src/read_cache.c
index 4af4056..9835647 100644
--- a/src/read_cache.c
+++ b/src/read_cache.c
@@ -88,7 +88,7 @@ struct read_cache_s {
 #  define dprintf(fmt, ...) fprintf(MSG_OUT, "libdvdnav: %s: "fmt,  __func__ , __VA_ARGS__)
 # else
 #ifdef _MSC_VER
-#  define dprintf(fmt, str) /* Nowt */
+#  define dprintf(fmt, ...) /* Nowt */
 #else
 #  define dprintf(fmt, ...) /* Nowt */
 #endif /* _MSC_VER */
