diff --git a/msvc/include/os_types.h b/msvc/include/os_types.h
index 294847e..9e30871 100644
--- a/msvc/include/os_types.h
+++ b/msvc/include/os_types.h
@@ -12,7 +12,6 @@ typedef signed   char			sint_32;
 typedef signed   short		sint_16;
 typedef signed   int			sint_8;
 
-#define snprintf _snprintf
 #define M_PI            3.14159265358979323846  /* pi */
 #define DLLENTRY __declspec(dllexport)
 
diff --git a/msvc/include/unistd.h b/msvc/include/unistd.h
index 1d420d0..0df90bc 100644
--- a/msvc/include/unistd.h
+++ b/msvc/include/unistd.h
@@ -29,6 +29,7 @@
 #include <malloc.h>
 #include <errno.h>
 #include <direct.h>
+#include <io.h>
 
 #include <config.h>
 
@@ -60,9 +61,6 @@
 #define strncasecmp _strnicmp
 #endif
 
-#define snprintf _snprintf
-#define vsnprintf _vsnprintf
-
 // FIXME : I dont remember why this is here
 #define readlink
 
