diff --git a/.vs/sys/time.h b/.vs/sys/time.h
new file mode 100644
index 0000000..36cecfb
--- /dev/null
+++ b/.vs/sys/time.h
@@ -0,0 +1,13 @@
+#ifndef _VS_SYS_TIME_H
+#define _VS_SYS_TIME_H
+#ifndef _WINSOCKAPI_
+#include <winsock.h>
+#endif
+#include <sys/timeb.h>
+static void gettimeofday(struct timeval* tv, void* tz) {
+    struct timeb tb;
+    ftime(&tb);
+    tv->tv_sec = tb.time;
+    tv->tv_usec = 1000 * tb.millitm;
+}
+#endif
diff --git a/.vs/unistd.h b/.vs/unistd.h
new file mode 100644
index 0000000..1cb1d6d
--- /dev/null
+++ b/.vs/unistd.h
@@ -0,0 +1,10 @@
+#ifndef _VS_UNISTD_H
+#define _VS_UNISTD_H
+#include <io.h>
+#include <fcntl.h>
+#include <windows.h>
+#define STDIN_FILENO 0
+#define STDOUT_FILENO 1
+#define STDERR_FILENO 2
+static void usleep(unsigned int us) { Sleep((us+999)/1000); }
+#endif
diff --git a/src/Makefile.am b/src/Makefile.am
index 4abedaf..44dca66 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -24,6 +24,7 @@ cd_paranoia_DEPENDENCIES =  $(LIBCDIO_DEPS) $(LIBCDIO_CDDA_LIBS) $(LIBCDIO_PARAN
 bin_PROGRAMS = cd-paranoia
 
 AM_CPPFLAGS = -I$(top_srcdir) $(LIBCDIO_PARANOIA_CFLAGS) $(LIBCDIO_CFLAGS)
+AM_CPPFLAGS += -Dalloca=_alloca
 
 cd-paranoia.$(OBJEXT): usage.h
 
