diff --git a/include/cdio/types.h b/include/cdio/types.h
index 1845259..f321694 100644
--- a/include/cdio/types.h
+++ b/include/cdio/types.h
@@ -59,7 +59,18 @@ typedef uint8_t ubyte;
    unistd.h that defines them. Such a file is provided with
    the libcdio source, in the MSVC/missing directory */
 #if defined(_MSC_VER)
-#include <unistd.h>
+#ifndef _MODE_T
+#define _MODE_T
+typedef unsigned short mode_t;
+#endif
+#ifndef _SSIZE_T
+#define _SSIZE_T
+#ifdef _WIN64
+typedef long long ssize_t;
+#else
+typedef long ssize_t;
+#endif
+#endif
 #endif
 
   /* default HP/UX macros are broken */
