diff --git a/src/Makefile.am b/src/Makefile.am
index ee46ba5..6ae7897 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -89,8 +89,9 @@ check_programs  += mmc-tool
 bin_PROGRAMS = $(bin_cd_drive) $(bin_cd_info)  $(bin_cdinfo_linux) $(bin_cd_read) $(bin_iso_info) $(bin_iso_read) $(bin_cdda_player) $(bin_mmc_tool)
 
 AM_CPPFLAGS = -I$(top_srcdir) $(LIBCDIO_CFLAGS) $(VCDINFO_CFLAGS) $(CDDB_CFLAGS)
+AM_CPPFLAGS += -Dalloca=_alloca
 
-man_MANS = $(man_cd_drive) $(man_cd_info) $(man_cd_read) $(man_iso_read) $(man_iso_info)
+man_MANS =
 EXTRA_DIST = cd-drive.help2man cd-info.help2man cd-read.help2man \
              iso-info.help2man iso-read.help2man $(GETOPT_C) getopt_int.h \
              $(man_MANS)
